src/pint/models/binary_piecewise.py

```diff
@@ -0,0 +1,465 @@
+
+            for i in range(len(lb)):
+                for j in range(len(lb)):
+                    if i != j:
+                        if max(lb[i], lb[j]) < min(ub[i], ub[j]):
+                            raise ValueError(
+                                f"Group boundary overlap detected. Make sure groups are not overlapping"
+                            )
+
+    def get_group_boundaries(self):
+        """Get a all pieces' date boundaries.
+        Returns
+        -------
+        list
+                np.array
+                        (length: toas) List of piecewise orbital parameter lower boundaries
+                np.array
+                        (length: toas) List of piecewise orbital parameter upper boundaries
+        """
+        # asks the object for the number of piecewise groups
+        return self.binary_instance.get_group_boundaries()
+
+    def which_group_is_toa_in(self, toa):
+        """Find the group a toa belongs to based on the boundaries of groups passed to BT_piecewise
+        Parameters
+        ----------
+        toa : toa
+           TOA/TOAs to check which group they're in
+        Returns
+        -------
+        np.array
+           str elements, look like ['0000','0001'] for two TOAs where one refences T0X_0000 or T0X_0001.
+        """
+        # asks the model what group a single TOA/list of TOAs
+        barycentric_toa = self._parent.get_barycentric_toas(toa)
+        return self.binary_instance.toa_belongs_in_group(barycentric_toa)
+
+    def get_number_of_groups(self):
+        """Get the number of piecewise parameters"""
+        return len(self.binary_instance.piecewise_parameter_information)
+
+    def get_group_indexes(self):
+        """Get all the piecewise parameter labels
+        Returns
+        -------
+        np.array
+                (length: number of piecewise groups) List of piecewise parameter labels e.g with pieces T0X_0000, T0X_0001, T0X_0003, returns [0,1,3]
+        """
+        group_indexes = []
+        for i in range(0, len(self.binary_instance.piecewise_parameter_information)):
+            group_indexes.append(
+                self.binary_instance.piecewise_parameter_information[i][0]
+            )
+        return group_indexes
+
+    def get_group_indexes_in_four_digit_format(self):
+        """Get all the piecewise parameter labels in 4 digit format
+        Returns
+        -------
+        np.array
+                (length: number of piecewise groups) List of piecewise parameter labels in 4 digit format e.g with pieces T0X_0000, T0X_0001, T0X_0003, returns [0000,0001,0003]
+        """
+        group_indexes = []
+        for i in range(0, len(self.binary_instance.piecewise_parameter_information)):
+            group_indexes.append(
+                f"{int(self.binary_instance.piecewise_parameter_information[i][0]):04d}"
+            )
+        return group_indexes
+
+    def get_T0Xs_associated_with_toas(self, toas):
```

Comment:
maybe instead of having specialized methods for `TOX` and `A1X`, you can have 1 method to see which group(s) TOAs are associated with?  Then the `TOX` and `A1X` can just be gotten using the existing mappings.

Reply:
implemented new function paramX_per_toa('T0',toa) which does as was described returns same as get_T0Xs_associated_with_toas. This function changes the structure of the testing suite since now a lot of convenience functions can be consolidated into this